rpcs3/util/sysinfo.cpp

```diff
@@ -177,6 +177,71 @@ bool utils::has_avx512_vnni()
 #endif
 }
 
+bool utils::has_avx10()
+{
+#if defined(ARCH_X64)
+	// Implies support for most AVX-512 instructions
+	static const bool g_value = get_cpuid(0, 0)[0] >= 0x7 && get_cpuid(7, 1)[3] & 0x80000;
+	return g_value;
+#else
+	return false;
+#endif
+}
+
+bool utils::has_avx10_512()
+{
+#if defined(ARCH_X64)
+	// AVX10 with 512 wide vectors
+	static const bool g_value = has_avx10() && get_cpuid(24, 0)[2] & 0x40000;
+	return g_value;
+#else
+	return false;
+#endif
+}
+
+u32 utils::avx10_isa_version()
+{
+#if defined(ARCH_X64)
+	// 8bit value
+	static const u32 g_value = []()
+	{
+		u32 isa_version = 0;
```

Comment:
Initialize to umax here.

Reply:
It's defined as >= 1, but umax might be more obvious

Reply:
Actually on second thought, that's a bad idea, if you wanted something from AVX10.5 or greater, you'd check for if (avx10_isa_version() >= 5), but if we initialize to umax then it'll pass the check.